Assets/Global/Scripts/Enemies/RangedEnemy/RangedEnemyStates/RangeChasingState.cs

```diff
@@ -0,0 +1,101 @@
+using UnityEngine;
+using UnityEngine.AI;
+
+namespace EnemiesNS
+{
+    public class RangeChasingState : BaseChasingState
+    {
+        private bool isMovingToRandomPosition = false;
+        public RangeChasingState(RangedEnemy enemy) : base(enemy) { }
+
+        public override void Enter()
+        {
+            enemy.animator.SetBool("Walk", true);
+            base.Enter();
+        }
+
+        public override void Exit()
+        {
+            enemy.animator.SetBool("Walk", false);
+            base.Exit();
+        }
+
+
+        private float timeSinceLastRandomMove = 0f;
+        private float randomMoveCooldown = 2f; // Cooldown time in seconds
+
+        public override void Update()
+        {
+            if (!enemy.isWaiting)
+            {
+                if (enemy.agent.isStopped)
+                    enemy.FreezeMovement(false);
+
+                if (isMovingToRandomPosition)
+                {
+                    // Check if the enemy has reached the random position
+                    if (!enemy.agent.pathPending && enemy.agent.remainingDistance <= enemy.agent.stoppingDistance)
+                    {
+                        isMovingToRandomPosition = false; // Reset the flag
+                        timeSinceLastRandomMove = Time.time; // Start cooldown
```

Comment:
are you using time since application start on purpose here, instead of Time.deltaTime?